harness/determined/deploy/local/cluster_utils.py

```diff
@@ -177,49 +153,103 @@ def master_up(
         )
         make_temp_conf = True
 
-    # Ensure checkpoint storage directory exists.
-    final_storage_host_path = master_conf.get("checkpoint_storage", {}).get("host_path")
-    if final_storage_host_path is not None:
-        final_storage_host_path = Path(final_storage_host_path)
-        if not final_storage_host_path.exists():
-            final_storage_host_path.mkdir(parents=True)
-
     if make_temp_conf:
         fd, temp_path = tempfile.mkstemp(prefix="det-deploy-local-master-config-")
         with open(fd, "w") as f:
             yaml.dump(master_conf, f)
-        master_config_path = Path(temp_path)
+        master_config_path = pathlib.Path(temp_path)
 
     # This is always true by now, but mypy needs help.
     assert master_config_path is not None
+    restart_policy = "unless-stopped" if autorestart else "no"
+    env = {
+        "INTEGRATIONS_HOST_PORT": str(port),
+        "DET_DB_PASSWORD": db_password,
+        "IMAGE_REPO_PREFIX": image_repo_prefix,
+        "DET_VERSION": version,
+        "DET_RESTART_POLICY": restart_policy,
+        "DET_MASTER_CONFIG": str(master_config_path.resolve()),
+    }
+
+    # Kill existing master container if exists.
+    master_down(master_name=master_name, delete_db=delete_db)
 
-    env["DET_MASTER_CONFIG"] = str(master_config_path.resolve())
+    # Create network.
+    docker_client.networks.create(NETWORK_NAME)
 
-    master_down(master_name, delete_db)
-    docker_compose(command, master_name, env)
+    # Start db.
+    db_up(password=db_password, network=NETWORK_NAME)
+
+    # Start master instance.
+    volumes = [f"{os.path.abspath(master_config_path)}:/etc/determined/master.yaml"]
+    docker_client.containers.run(
+        image=f"{image_repo_prefix}/determined-master:{version}",
+        environment=env,
+        init=True,
+        mounts=[],
+        volumes=volumes,
+        name=master_name,
+        detach=True,
+        labels={},
+        restart_policy={"Name": restart_policy},
+        device_requests=None,
+        ports={f"{port}": "8080"},
+        network=NETWORK_NAME,
+    )
     _wait_for_master("localhost", port, cluster_name)
 
 
+def db_up(password: str, network: str) -> None:
+    env = {"POSTGRES_DB": "determined", "POSTGRES_PASSWORD": password}
+
+    docker_client.containers.run(
+        image="postgres:10.14",
+        environment=env,
+        init=True,
+        mounts=[],
+        volumes=[f"{VOLUME_NAME}:/var/lib/postgresql/data"],
+        name=DB_NAME,
+        detach=True,
+        network=network,
+        labels={},
+        restart_policy={"Name": "unless-stopped"},
+        device_requests=None,
+    )
+
+
 def master_down(master_name: str, delete_db: bool) -> None:
+    # Kill master instance.
+    _kill_container(master_name)
+
+    # Remove existing db container if exists.
+    _kill_container(DB_NAME)
```

Comment:
I happen to know that if you sigkill the database, postgres increments all of its autoincrement counters by 32.  Can we sigterm the database instead?  Postgres seems to like that a lot better.